### PR TITLE
Add Object-based querying support and field order fix

### LINKED ISSUE #111: Set column order in dataset query DataFrame from the dataset fields list order
Querying datasets from R produces a DataFrame with arbitrarily-ordered fields. The order of the fields should be set from the field ordering values which can be retrieved from the API.

### LINKED ISSUE #113: Add file-based querying support to Object resource
Some files (JSON/CSV/TSV) can be directly queried. Add support for this to the Object resource.

### PR DESCRIPTION
**Pull request by davecap**
Adds Object.data(), Object.query(), and Object.fields() methods
closes #113

Retrieves fields in order and sets row.names of the dataframe
closes #111

### CODE REVIEW

**Comment by davecap** [comment]
@nikolamaric please review/approve the code changes when you get a chance

**Review by nikolamaric — DISMISSED** [review]
All looks good. Tested with a few files from my vault.

File: R/object.R [hunk 1]
```diff
@@ -373,3 +373,141 @@ Object.get_or_upload_file <- function(local_path, vault_id, vault_path, filename
...
+#'
+#' @examples \dontrun{
+#' Object.fields("1234567890")
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+Object.fields <- function(id, env = solvebio:::.solveEnv, ...) {
```
Comment (davecap) [inline R/object.R:499]: @nikolamaric do you think we should return the response (as currently works) or return the contents of `$fields`?

Comment (nikolamaric) [reply R/object.R:499]: In the python client we return list of field, maybe we can do the same here too?

Comment (davecap) [reply R/object.R:499]: Do you want to try to make that change and test it? The docs would need an update for that fields example.

Comment (nikolamaric) [reply R/object.R:499]: Sure. Will do both tomorrow.

File: R/dataset.R [hunk 2]
```diff
@@ -221,12 +221,15 @@ Dataset.fields <- function(id, env = solvebio:::.solveEnv, ...) {
     if (missing(id)) {
         stop("A dataset ID is required.")
     }
-    if (class(id) == "Dataset" || class(id) == "Object") {
+
+    if (class(id) == "Dataset") {
```
Comment (davecap) [inline R/dataset.R]: Technically object is ok actually, if they pass in a Logical Object ID

File: R/dataset.R [hunk 3]
```diff
@@ -221,12 +221,15 @@ Dataset.fields <- function(id, env = solvebio:::.solveEnv, ...) {
...
         id <- id$id
+    } else {
+        warning("A given object is not a dataset.")
     }
 
     path <- paste("v2/datasets", paste(id), "fields", sep="/")
-    .request('GET', path=path, query=list(...), env=env)
+    .request('GET', path=path, query=list(...), env=env)$fields
```
Comment (davecap) [inline R/dataset.R]: Can you check if Dataset.fields() is called from anywhere in the code? this is actually a breaking change... maybe we should just leave it

File: R/object.R [hunk 4]
```diff
@@ -504,10 +504,13 @@ Object.fields <- function(id, env = solvebio:::.solveEnv, ...) {
...
+
     if (class(id) == "Object") {
         id <- id$id
+    } else {
+        warning("A given object is not a file.")
     }
 
     path <- paste("v2/objects", paste(id), "fields", sep="/")
-    .request('GET', path=path, query=list(...), env=env)
+    .request('GET', path=path, query=list(...), env=env)$fields
```
Comment (davecap) [inline R/object.R:513]: check line https://github.com/solvebio/solvebio-r/pull/114/files/a51afeefd1537b1dea31b3dbceec4a3cfa7afc81..3877315beda7b83d535415336a770dff9b22c24d#diff-79d3aeda07361e6ff4c03f300d5c9ea2929e10d95b039b2ebeca51f5875e7547R457

**Review by nikolamaric — APPROVED** [review]

**Pull request merged**